File: backend/src/main/java/com/approval/service/NotificationService.java
package com.approval.service;

import com.approval.vo.NotificationVO;
import com.baomidou.mybatisplus.core.metadata.IPage;

/**
 * 通知服务接口
 */
public interface NotificationService {

    /**
     * 发送审批通知
     *
     * @param approvalId 审批ID
     * @param receiverId 接收人ID
     * @param title      通知标题
     * @param content    通知内容
     */
    void sendApprovalNotification(String approvalId, Long receiverId, String title, String content);

    /**
     * 发送系统通知
     *
     * @param receiverId 接收人ID
     * @param title      通知标题
     * @param content    通知内容
     */
    void sendSystemNotification(Long receiverId, String title, String content);

    /**
     * 获取用户通知列表
     *
     * @param userId   用户ID
     * @param page     页码
     * @param pageSize 每页条数
     * @param isRead   是否已读筛选（可选）
     * @return 分页结果
     */
    IPage<NotificationVO> getNotifications(Long userId, int page, int pageSize, Boolean isRead);

    /**
     * 标记单条通知为已读
     *
     * @param id     通知ID
     * @param userId 用户ID
     */
    void markAsRead(String id, Long userId);

    /**
     * 标记所有通知为已读
     *
     * @param userId 用户ID
     */
    void markAllAsRead(Long userId);

    /**
     * 获取未读通知数量
     *
     * @param userId 用户ID
     * @return 未读数量
     */
    int getUnreadCount(Long userId);
}

File: frontend/src/services/notificationService.ts
/**
 * 通知服务
 *
 * 处理通知相关API调用
 */

import api from './api'
import type { ApiResponse, PaginatedData } from '@/types'

/**
 * 通知
 */
export interface Notification {
    /** 通知ID */
    id: string
    /** 通知标题 */
    title: string
    /** 通知内容 */
    content: string
    /** 通知类型 */
    type: 'APPROVAL' | 'SYSTEM' | 'REMINDER'
    /** 关联业务ID */
    relatedId?: string
    /** 是否已读 */
    isRead: boolean
    /** 阅读时间 */
    readAt?: string
    /** 创建时间 */
    createdAt: string
}

/**
 * 获取通知列表
 *
 * @param page 页码
 * @param pageSize 每页条数
 * @param isRead 是否已读筛选（可选）
 * @returns 分页结果
 */
export async function getNotifications(
    page: number = 1,
    pageSize: number = 10,
    isRead?: boolean
): Promise<PaginatedData<Notification>> {
    const params: Record<string, unknown> = { page, pageSize }
    if (isRead !== undefined) {
        params.isRead = isRead
    }
    const response = await api.get<ApiResponse<PaginatedData<Notification>>>('/v1/notifications', { params })
    return response.data.data
}

/**
 * 标记单条通知为已读
 *
 * @param id 通知ID
 */
export async function markAsRead(id: string): Promise<void> {
    await api.put<ApiResponse<void>>(`/v1/notifications/${id}/read`)
}

/**
 * 标记所有通知为已读
 */
export async function markAllAsRead(): Promise<void> {
    await api.put<ApiResponse<void>>('/v1/notifications/read-all')
}

/**
 * 获取未读通知数量
 *
 * @returns 未读数量
 */
export async function getUnreadCount(): Promise<number> {
    const response = await api.get<ApiResponse<{ count: number }>>('/v1/notifications/unread-count')
    return response.data.data.count
}
